### PR TITLE
sessions controller and services setup for github authorization

### PR DESCRIPTION
**Pull request by ianderse**

### CODE REVIEW

**Comment by ianderse** [comment]
IMPORTANT: Everyone will also need to create a .env file in your application that contains the github secret keys. I'll send those in slack channel.

File: app/services/user_factory.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+  end
+
+  def find_or_create_user
+    User.find_or_create_by(name: authenticator.name)
+  end
+
+  private
+
+  attr_reader :authenticator
+end
```
Comment (ianderse) [inline app/services/user_factory.rb]: I have it changed on my local copy but it's not pushing up the revised version with the newline added, not sure why.

File: .env [hunk 2]
```diff
@@ -0,0 +1,2 @@
+GITHUB_KEY=06116e8c585f9852b942
```
Comment (r00k) [inline .env:1]: These keys are now compromised and need to be reset.

File: config/secrets.yml [hunk 3]
```diff
@@ -12,6 +12,8 @@
 
 development:
```
Comment (r00k) [inline config/secrets.yml:13]: This file should not be in source control.

Literally ALL your secret keys are now public.

**Pull request merged**